zipline/_protocol.pyx

```diff
@@ -21,6 +21,28 @@ from cpython cimport bool
 
 from zipline.assets import Asset
 
+
+class assert_keywords(object):
+    """
+    Asserts that the keywords passed into the wrapped function are included
+    in those passed into this decorator. If not, raise a TypeError with a
+    meaningful message, unlike the one cython returns by default.
+    """
+
+    def __init__(self, *args):
+        self.names = args
+
+    def __call__(self, func):
```

Comment:
Double checked by comparing this branch against `lazy-mainline` with an algo that uses `data.history(context.stocks, 'high', 3, '1d')` and at least in that case, this wrapper seems to be in the clear as far as adding overhead.